backend/frontend/src/screens/RegisterScreen.js

```diff
@@ -25,47 +29,56 @@ const style = {
 };
 
 const linkStyle = {
-  textDecoration: 'underline',
-  color : 'purple',
-  cursor : 'pointer'
-}
+  textDecoration: "underline",
+  color: "purple",
+  cursor: "pointer",
+};
 
 function RegisterScreen() {
   const navigate = useNavigate();
-  // const location = useLocation();
   const dispatch = useDispatch();
   const [name, setName] = useState("");
   const [email, setEmail] = useState("");
+  const [validatedEmail, setValidatedEmail] = useState("");
   const [password, setPassword] = useState("");
   const [phoneNumber, setPhoneNumber] = useState("");
   const [confirmPassword, setConfirmPassword] = useState("");
-  // const [message, setMessage] = useState("");
   const [openForm, setOpenForm] = useState(false);
   const [loader, setLoader] = useState(false);
   const [otp, setOtp] = useState("");
-
-  // const redirect = location.search ? location.search.split("=")[1] : "/";
+  const [submit, setSubmit] = useState(false);
 
   const { registerError, registerUserLoading } = useSelector(
     (state) => state.userRegister
   );
 
   const { userInfo } = useSelector((state) => state.userLogin);
   const { otpLoading } = useSelector((state) => state.generateOtp);
+  const { userValidate, userValidateError } = useSelector((state) => state.userValidator);
 
-  useEffect(() => {
-    setOpenForm(false);
-    if (userInfo) {
-      toast.success("user signup sucess!");
-      navigate("/");
-    } else if (registerError) {
-      toast.error(registerError);
-      setOpenForm(false);
-      dispatch({
-        type: USER_LOGOUT,
-      });
+  const otpGenerate = () => {
+    setOpenForm(true);
+    dispatch(generateOTP(email));
+  };
+
+  const regenerateOtp = () => {
+    setLoader(true);
+    dispatch(generateOTP(email));
+  };
+
+  const validateEmail = (e) => {
+    e.preventDefault();
+    if (password.length < 8) {
+      toast.error("pasword must be atleast 8 characters");
```

Comment:
First letter is not capital, but in next toast, it is in capital, so let's try to follow a single way of added errors.